todo-backend/README.adoc

```diff
@@ -0,0 +1,452 @@
+of prefixes we used for each individual filter (such as `acam` in the example above):
+
+* `FILTERS=acao, acam, acah, acac, acma`
+
+ifndef::ProductRelease[]
+
+== Run the Backend Locally
+
+=== Package the Backend as a Bootable Jar
+
+The backend is packaged as a Bootable Jar and configured to be deployable on OpenShift with the Maven Profile `bootable-jar-openshift`:
+
+[source,options="nowrap"]
+----
+$ mvn clean package -P bootable-jar-openshift
+----
+
+=== Run a Local PostgreSQL Database
+
+Before running the backend locally, we need to have a local PostgreSQL database that we can connect to.
+We use the `postgresql` docker image to create one:
+
+[source,options="nowrap"]
+----
+$ docker run --name todos-db \
+  -e POSTGRES_USER=todos \
+  -e POSTGRES_PASSWORD=mysecretpassword \
+  -p 5432:5432 \
+  postgres
+----
+
+This will create a database named `todos-db` that we can connect to on `localhost:5432` with the credentials `todos / mysecretpassword`.
+
+=== Run the Bootable Jar Application
+
+With the PostgreSQL database running, we can start the backend by passing the required environment variables to connect to the database:
+
+[source,options="nowrap"]
+----
+$ POSTGRESQL_DATABASE=todos \
+  POSTGRESQL_SERVICE_HOST=localhost \
+  POSTGRESQL_SERVICE_PORT=5432 \
+  POSTGRESQL_USER=todos \
+  POSTGRESQL_PASSWORD=mysecretpassword \
+  POSTGRESQL_DATASOURCE=ToDos \
+  java -jar target/todo-backend-bootable.jar
+...
+14:41:58,111 INFO  [org.jboss.as.server] (Controller Boot Thread) WFLYSRV0010: Deployed "todo-backend.war" (runtime-name : "ROOT.war")
+...
+----
+
+The backend is running, and we can use the HTTP API to manage a list of todos:
+
+[source,options="nowrap"]
+----
+# get a list of todos
+$ curl http://localhost:8080
+[]
+
+# create a todo with the title "This is my first todo item!"
+$ curl -X POST -H "Content-Type: application/json"  -d '{"title": "This is my first todo item!"}' http://localhost:8080/
+{"completed":false,"id":1,"order":0,"title":"This is my first todo item!","url":"https://localhost:8080/1"}%
+
+# get a list of todos with the one that was just created
+$ curl http://localhost:8080
+[{"completed":false,"id":1,"order":0,"title":"This is my first todo item!","url":"https://localhost:8080/1"}
+----
+endif::[]
+
+== Run the Backend on OpenShift
+
+=== Prerequisites
+
+* You must be logged in OpenShift and have an `oc` client to connect to OpenShift
+* https://helm.sh[Helm] must be installed to deploy the backend on OpenShift.
+
+Once you have installed Helm, you need to add the repository that provides Helm Charts for {productName}:
+
+ifndef::ProductRelease[]
+[source,options="nowrap"]
+----
+$ helm repo add wildfly https://docs.wildfly.org/wildfly-charts/
+"wildfly" has been added to your repositories
+$ helm search repo wildfly
+NAME                    CHART VERSION   APP VERSION     DESCRIPTION
+wildfly/wildfly         1.1.0           22.0            Build and Deploy WildFly applications on OpenShift
```

Comment:
I know this is only an example of the output but what about to parameterize the versions here?
Although I don't know how that works and will work for future WFs, there seems to be no branches on that repo

Reply:
I've added ellipsis for the command output (`...`), the version are not meaningful, it just shows how to install the chart repository.